OneCommunityGlobal/HighestGoodNetworkApp
# Sucheta Replace the blue square self-scheduler for anyone with 5 or more blue squares (anyone other than Owner/Admin)

# Description
(PRIORITY HIGH) Jae: (WIP Sucheta ) Replace the blue square self-scheduler for anyone with 5 or more blue squares (**anyone other than Owner/Admin**)
- Related PRs [PR1542](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1542)+614, [PR1081](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1081) 
- Profile Page>Below Blue Squares 
- Replace with something that is a different color (not red) that says
Can’t Schedule Time Off
(click to learn why)
- This is needed because people with 5 or more blue squares should no longer be on the team and, if they still are, will need an Admin to enter their time off for them
- It should account for any future scheduled time off too (see above)
- When they click the new button, it should say this if 5 or more blue squares:

*You have X blue squares already and 5 are the maximum allowed per year. Please contact your Administrator if you need to request time off. 

Note: Blue squares drop off 1 calendar year after they are issued. *

- When they click the new button, it should say this if they have scheduled time off equaling 5 or more blue squares:

*Including your time already requested off, you have used the equivalent of X blue squares. 5 is the maximum allowed per year. Please remove a time-off request below or contact your Administrator if you need to request time off in addition to what is listed here:

Time off request 1 
Time off request 2
Etc.

Note: Blue squares drop off 1 calendar year after they are issued. *

## Related PRS (if any):
No backend PRs
…

## Main changes explained:

1. Created component: 
- components > UserProfile > SchedulerExplanationModal 

2. Adds new handler functions to BlueSquareLayout.jsx, that opens modals based on user roles and the number of BlueSquares assigned or scheduled.

3. The button `Schedule Blue Square Reason` changes to `Can't Schedule Time Off` if `((isInfringementMoreThanFive || numberOfReasons >= 5 || (infringementsNum + numberOfReasons >= 5 )) && !(userProfile.role === "Administrator" || userProfile.role === "Owner"))` 


…

## How to test:
1. check into the current branch
2. do `npm install` and `...` to run this PR locally
5. Clear site data/cache
6. log as (volunteer/ manager/ assistant manager) user
7. go to dashboard→ View Profile → BlueSquare section →…
8. If the numbers of BlueSquare are less than 5 the Users with permission to Schedule blueSquares, should see the `Schedule Blue Square` button. Otherwise, the button changes to `Can't Schedule Time Off Click to learn why`.
9. Click on the button to see a modal with an explanation of why scheduling has been disabled.

## Screenshots or videos of changes:
**Before Change**
https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/101291283/159ce01e-da12-4b2a-ac37-86c47a8e26cb

**Updated Changes**



https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/101291283/7c12b59f-39d5-4bfa-a98e-446947cb3520



## Note:
Please note nothing has changed for the user role Admin/Owner. **Last updated 01/04/2024**
